Frameworks/MQTTnet.NetStandard/Implementations/MqttCommunicationAdapterFactory.cs

```diff
@@ -21,6 +21,11 @@ public IMqttCommunicationAdapter CreateMqttCommunicationAdapter(MqttClientOption
                 return new MqttChannelCommunicationAdapter(new MqttWebSocketChannel(webSocketOptions), new MqttPacketSerializer { ProtocolVersion = options.ProtocolVersion });
             }
 
+            if (options is MqttClientQueuedOptions queuedOptions)
```

Comment:
I want' to avoid such things (see the long comment for details below). In my opinion the new client should be separated from all other stuff of the already existing one. It should be like an extension. Like if you crate your own nuget package with this implementation having this packet only as a reference. The overall client factory is OK but other things should not be touched.

Reply:
muble, muble, muble.
I do not agree. If I was a programmer,  I would like a complete client with all stuff like other clients in Java and C++. All other C# clients lacks some features, this one would the first with all the suff needed, and you can choose which one to use.

Reply:
I absolutely agree here. But the question is why the base implementation must be changed. This is not clear to me. I still see the new client as an extension which adds functionality like queuing or retrying. In my opinion there is no need to change the already existing code in any way.  Also because your implementation here is simply wrong. You check for your (new) options and always return a TCP channel. So a user which requires WebSockets cannot use the queued client. There are also other places which have to be changed if you follow this strategy. This why I am saying lets treat it as a completely separate client. Then the chance is very small that we forget to support some combinations. Also the consideration or a channel (TCP, WS) should not be part of the new queued client. This should be still the job ob the low level client. The client factory is a separate story and the only place were I can accept it because this is the entry point for the users. There they can decide. Probably a unsafe cast is required here like `options as MqttClientTcpOptions` instead of the check with `is`. Then you can pass in the inherited options and it will still work. I also want to refer to the comment where I mentioned interfaces for the options. But to keep it short. You can also complete your features then I will merge it and rework some parts to fit his requirements. If it is OK for you.

Reply:
Yes please. I lost you here: "You check for your (new) options and always return a TCP channel. So a user which requires WebSockets cannot use the queued client".
You want the queued client to be used with WS too. I agree. But you do not want to change existing code? Do not know how to do that. Moreover I changed some exisiting code transfomingh Private methds in Internal methods.
Sure I'm missing something. I'm committing the changes I understand. Would be better if you can continue with WS. I'm thinking about a better solution for Options.
But you, of course, have a better and complete vision.

Reply:
OK for me it is OK. Let me implement this. I will merge this pull request because I still want to honor your work and ideas. So you will be shown as an official contributor. Please wait with all until I committed these changes. I will implement this today in the evening. The can start a new discussion if this fits your needs and what needs to be changed. Do you agree?

Reply:
OK!